README.md

```diff
@@ -408,6 +365,90 @@ final class UserRoutes: RouteCollection {
 | `method` | The method of the route.                 |
 | `path`   | The path for the route.                  |
 
+## Tests
+These templates are related to unit testing with XCTest.
+
+### LinuxMain
+Generates a static `allTests` for every `XCTestCase` and registers them in the `LinuxMain.swift`.
+
+Example:
+
+```swift
+class UserControllerTests: TestCase {
+    func testShowOneUser() throws {}
+    func testShowAllusers() throws {}
+}
+```
+
+Becomes:
+
+```swift
+#if os(Linux)
+
+import XCTest
+@testable import MyProjectTests
+
+// sourcery:inline:auto:LinuxMain
+
+extension UserControllerTests {
+  static var allTests = [
+    ("testShowOneUser", testShowOneUser),
+    ("testShowAllusers", testShowAllusers),
+  ]
+}
+
+XCTMain([
+  testCase(UserControllerTests.allTests)
+])
+
+// sourcery:end
+
+#endif
+```
+
+#### Annotations
+
+| Key                    | Description                              |
+| ---------------------- | ---------------------------------------- |
+| `excludeFromLinuxMain` | Prevents the test case from being included in the generated code. |
+
+## General
+
+These templates are not related to any specific part of your Vapor project.
+
+### Enums
+
+Generates convenient accessors for getting a list of all cases. The MySQL preparation will use this as well.
+
+Example:
+
+```swift
+TODO
```

Comment:
Let's get these filled in as well. Also I think we might have to clean up some old enum annotations in the readme.

Reply:
The controller section with how routes are being generated seems a bit outdated as well